ts/screens/profile/ProfileDataScreen.tsx

```diff
@@ -1,96 +1,144 @@
-import * as React from "react";
-import { connect } from "react-redux";
+import React, { ComponentProps, useCallback, useMemo } from "react";
 import { ContextualHelpPropsMarkdown } from "../../components/screens/BaseScreenComponent";
-import ListItemComponent from "../../components/screens/ListItemComponent";
-import { RNavScreenWithLargeHeader } from "../../components/ui/RNavScreenWithLargeHeader";
+import { IOScrollViewWithLargeHeader } from "../../components/ui/IOScrollViewWithLargeHeader";
 import I18n from "../../i18n";
 import { useIONavigation } from "../../navigation/params/AppParamsList";
 import ROUTES from "../../navigation/routes";
 import {
   hasProfileEmailSelector,
-  isProfileEmailValidatedSelector,
   profileEmailSelector,
+  profileFiscalCodeSelector,
   profileNameSurnameSelector
 } from "../../store/reducers/profile";
-import { GlobalState } from "../../store/reducers/types";
+import { useIOSelector } from "../../store/hooks";
+import { FAQsCategoriesType } from "../../utils/faq";
+
+type EndElementProps = ComponentProps<typeof ListItemInfo>["endElement"];
 
-type Props = ReturnType<typeof mapStateToProps>;
+const FAQ_CATEGORIES: ReadonlyArray<FAQsCategoriesType> = [
+  "profile",
+  "privacy",
+  "authentication_SPID"
+];
 
 const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
   title: "profile.preferences.contextualHelpTitle",
   body: "profile.preferences.contextualHelpContent"
 };
 
-const ProfileDataScreen: React.FC<Props> = ({
-  profileEmail,
-  isEmailValidated,
-  hasProfileEmail,
-  nameSurname
-}): React.ReactElement => {
-  const navigation = useIONavigation();
-
-  const navigateToInsertEmailScreen = React.useCallback(() => {
-    navigation.navigate(ROUTES.PROFILE_NAVIGATOR, {
-      screen: ROUTES.INSERT_EMAIL_SCREEN,
-      params: {
-        isOnboarding: false
-      }
-    });
-  }, [navigation]);
+const ProfileDataScreen = () => {
+  const { navigate } = useIONavigation();
+  const profileEmail = useIOSelector(profileEmailSelector);
+  const hasProfileEmail = useIOSelector(hasProfileEmailSelector);
+  const nameSurname = useIOSelector(profileNameSurnameSelector);
+  const fiscalCode = useIOSelector(profileFiscalCodeSelector);
 
-  const onPressEmail = () => {
+  const onPressEmail = useCallback(() => {
     if (hasProfileEmail) {
-      navigateToInsertEmailScreen();
+      navigate(ROUTES.PROFILE_NAVIGATOR, {
+        screen: ROUTES.INSERT_EMAIL_SCREEN,
+        params: {
+          isOnboarding: false
+        }
+      });
     }
-  };
+  }, [hasProfileEmail, navigate]);
+
+  const email = useMemo(
+    () =>
+      pipe(
+        profileEmail,
+        O.getOrElse(() => I18n.t("global.remoteStates.notAvailable"))
+      ),
+    [profileEmail]
+  );
+
+  const showFiscalCode = useMemo<EndElementProps>(() => {
+    const label = I18n.t("global.buttons.show");
+
+    return {
+      type: "buttonLink",
+      componentProps: {
+        onPress: () => {
```

Comment:
To optimise, you could put the navigation you are using at onPress inside a function that uses the useCallback, what do you think?

Reply:
Since the entire object is memoized, the function within it is memoized too. So, wrapping the function with a `useCallback` is not needed for this use case!